### ISSUE TITLE
map containers larger than 4096px are scaled

### ISSUE THREAD

**Issue by andrewharvey** [mapbox/mapbox-gl-js#2673]
**mapbox-gl-js version**: 0.19.1

### Steps to Trigger Behavior

 1. Initialise a simple map (eg. https://www.mapbox.com/mapbox-gl-js/examples/) but set the HTML container to a width & height of 8192x8192px.
demo: http://jsbin.com/hadiravumo/1/edit?html,output

### Expected Behavior

I expect the map to be crisp and not over scaled. This is what the  4096x4096px map looks like, nice and crisp, and what I expect the 8192x8192px map to look like (just a larger area as the zoom is the same in both cases).

![4096](https://cloud.githubusercontent.com/assets/117278/15804613/941acd0a-2b52-11e6-82fe-596bceee2da9.png)

### Actual Behavior
It's blurry because it is rendered at 4096x4096px map but scaled by 2x. The 8192x8192px map looks like this:

![8192](https://cloud.githubusercontent.com/assets/117278/15804616/97becea2-2b52-11e6-9786-ffa9a743d73e.png)

```
var gl = canvas.getContext('experimental-webgl');
console.log(gl.getParameter(gl.MAX_RENDERBUFFER_SIZE));
```

returns 8192 which indicates that WebGL canvases that big should work? `window.devicePixelRatio` is 1.

**Comment by mourner** [comment]
I don't think we can do anything about this in GL JS code if it's a browser limitation. Apparently max renderbuffer size translates to half of that in pixels.

**Comment by andrewharvey** [comment]
>I don't think we can do anything about this in GL JS code if it's a browser limitation. Apparently max renderbuffer size translates to half of that in pixels.

@mourner Thanks for the explanation, I didn't realise that gl.MAX_RENDERBUFFER_SIZE translates to half of it in pixels, so I thought it was an issue in GL JS. I'm interested why it translates to half.

But given that, I'll have to go ahead with a workaround for my specific use case of generating high-res images for print by pasting together 4096x4096 tiles.

**Comment by mourner** [comment]
That's just a guess, — I don't know for sure. Let's reopen if this turns out to be something that can be fixed on our side.

**Comment by lucaswoj** [comment]
Let's reopen _if this turns out to be something that can be fixed on our side_.

**Comment by andrewharvey** [comment]
From https://github.com/mapbox/mapbox-gl-js/issues/3915#issuecomment-272987981

> I’m unable to get #2673 to reproduce in v0.28.0

@1ec5 I can confirm this. For me on Chrome 0.19.1 still suffers from this issue, but it appears fixed in 0.28.0. Which is awesome!

My gl.MAX_RENDERBUFFER_SIZE is still 8192 which is the size of the map I used to test.
